web/docs/deployment/ci-cd.md

```diff
@@ -1,7 +1,113 @@
 ---
 title: CI/CD Scenarios
 ---
+Setting up a CI/CD pipeline is an optional but highly recommended part of deploying applications.
 
-TODO: we'll talk about how to e2e test Wasp apps in the CI
+**Continuous Integration (CI)** involves verifying/testing code changes through an automated process whenever code is pushed to the repository. This helps us catch bugs early and make sure that our app works.
 
-TODO: we'll talk about building the server as a Docker image in the CI
\ No newline at end of file
+**Continuous Deployment (CD)** refers to the automatic deployment of code changes to the production environment. This is commonly know as "push to deploy" and frees developers from having to manually deploy code changes.
+
+## Running tests in CI
+
+### End to end tests
+
+End to end (e2e) tests simulate real user using your app and you can test different scenarios like login, adding items to cart, etc. Writing end to end tests frees you from
+manually testing your app after every change.
+
+**To run e2e tests with Wasp in the CI**, you'll need to:
+1. Install Wasp in the CI environment.
+2. Run your app (with the database) in the CI environment.
+3. Run the e2e tests against the running app.
+
+#### Example app
+
+We'll show you how to run end-to-end tests in CI using the [Github Actions](https://github.com/features/actions) as our CI and the [Playwright](https://playwright.dev/) as our e2e testing framework.
+
+1. Check our example app and its e2e tests in the [e2e-tests](https://github.com/wasp-lang/e2e-test-example/tree/main/e2e-tests) directory. 
+
+    You can copy the `e2e-tests` directory to your own project and modify it to fit your app. This will enable you to run the e2e tests locally.
+    
+    <details>
+    <summary>Example e2e test</summary>
+
+    ```ts
+    import { expect, test } from "@playwright/test";
+    import { generateRandomUser, logUserIn } from "./utils";
+
+    const user = generateRandomUser();
+
+    test.describe("basic user flow test", () => {      
+      test("log in and add task", async ({ page }) => {
+        await logUserIn({ page, user });
+        await expect(page).toHaveURL("/");
+        await expect(page.locator("body")).toContainText("No tasks yet.");
+
+        // Add a task
+        await page.fill('input[name="description"]', "First task");
+        await page.click('input:has-text("Create task")');
+        await expect(page.locator("body")).toContainText("First task");
+      });
+    });
+    ```
+    </details>
```

Comment:
You can probably drop this? It is just a small snippet, and it risks becoming out of date, we have to maintain it, real code is anyway in that example app you linked to.